src/Umbraco.Cms.Integrations.Search.Algolia/Notifications/NewContentPublishedHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Extensions.Logging;
+
+using System.Text.Json;
+
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Integrations.Search.Algolia.Builders;
+using Umbraco.Cms.Integrations.Search.Algolia.Migrations;
+using Umbraco.Cms.Integrations.Search.Algolia.Models;
+using Umbraco.Cms.Integrations.Search.Algolia.Services;
+
+namespace Umbraco.Cms.Integrations.Search.Algolia.Notifications
+{
+    public class NewContentPublishedHandler : INotificationHandler<ContentPublishedNotification>
```

Comment:
I think - though you'll have to check - that these notification handlers can be asynchronous.  If so, you can make this one so, and the calls it makes to upstream components where you are currently having to do a `GetAwaiter().GetResult()` or similar, can be converted to `await`.